src/services/transaction.ts

```diff
@@ -53,9 +54,74 @@ export const createTransaction = async (
  * @param transaction - 更新する取引
  * @returns 更新された取引
  */
-export const updateTransaction = async (transaction: Transaction) => {
+export const updateTransaction = async (transaction: Partial<Transaction>) => {
+  if (!transaction.id) throw new Error("idが指定されていません");
   return prisma.transaction.update({
     where: { id: transaction.id },
     data: transaction,
   });
 };
+
+/**
+ * 取引コメントを取得
+ * @param transactionId 取引ID
+ * @returns コメントの配列
+ */
+export const getTransactionComments = async (transactionId: string) => {
+  const comments = await prisma.transactionComment.findMany({
+    where: { transactionId },
+    include: { user: true },
```

Comment:
mapで再展開しなくても、以下のように書けば戻り値にユーザーのレコードを含めることができます。

```suggestion
include:{
    user:{
        select:{
            name:true,
            image:true,
            id:true
        }
    }
}
```